### DISCUSSION TITLE
catchError's error type should be unknown instead of any

### DISCUSSION THREAD

**Discussion by abdulkareemnalband** [ReactiveX/rxjs#7121] [Ideas / Feature request]
In best case scenario  catchError's callback's `error` as parameter should be typed as `unknown` instead as `any`
It will force users to handle error more robustly

Its similar to https://www.typescriptlang.org/tsconfig#useUnknownInCatchVariables

But simply changing `any` to `unknown` is major breaking change. And we also need a proper plan how to do that

It is not limited to `catchError` we also need to do the same for `retryWhen` , `subscribe` , `tap` and other `operators` and `functions` where we need to work with errors

**Comment by kaluginserg** [comment]
Previous discussions: #4317, #2646, #6328
